corehq/apps/app_manager/models.py

```diff
@@ -2226,6 +2270,88 @@ def update_app_case_meta(self, app_case_meta):
                 meta.add_closer(self.unique_id, action.close_condition)
 
 
+class SchedulePhaseForm(IndexedSchema):
+    """
+    A reference to a form in a schedule phase.
+    """
+    form_id = FormIdProperty("modules[*].schedule_phases[*].forms[*].form_id")
+
+
+class SchedulePhase(IndexedSchema):
+    """
+    SchedulePhases are attached to a module.
+    A Schedule Phase is a grouping of forms that occur within a period and share an anchor
+    A module should not have more than one SchedulePhase with the same anchor
+
+    anchor:                     Case property containing a date after which this phase becomes active
+    forms: 			The forms that are to be filled out within this phase
+    """
+    anchor = StringProperty()
+    forms = SchemaListProperty(SchedulePhaseForm)
+
+    @property
+    def id(self):
+        """ A Schedule Phase is 1-indexed """
+        _id = super(SchedulePhase, self).id
+        return _id + 1
+
+    @property
+    def phase_id(self):
+        return "{}_{}".format(self.anchor, self.id)
+
+    def get_module(self):
+        return self._parent
+
+    _get_forms = IndexedSchema.Getter('forms')
+
+    def get_forms(self):
+        """Returns the actual form objects related to this phase"""
+        module = self.get_module()
+        return (module.get_form_by_unique_id(form.form_id) for form in self._get_forms())
+
+    def get_form(self, desired_form):
```

Comment:
yeah... it's a bit strange, but this returns the form within the phase (with it's `id` within the phase that form is in)